app/workers/package_manager_download_worker.rb

```diff
@@ -65,7 +65,12 @@ def perform(platform_name, name, version = nil)
     raise "Platform '#{platform_name}' not found" unless platform
 
     # need to maintain compatibility with things that pass in the name of the class under PackageManager module
-    logger.info("Package update for platform=#{key} name=#{name}")
-    platform.update(name)
+    if platform::SUPPORTS_SINGLE_VERSION_UPDATE && version.present?
+      logger.info("Version update for platform=#{key} name=#{name} version=#{version}")
+      platform.update_version(name, version)
```

Comment:
What if update_version just knew how to handle this internally instead of branching here and exposing a constant?

Reply:
And the signatures all change to `platform.update(name, version = nil)` 

I s'pose that's a lot more files to touch though.